python/tk_multi_loader/model_item_data.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+from sgtk import TankError
+from sgtk.platform.qt import QtCore
+
+shotgun_model = sgtk.platform.import_framework("tk-framework-shotgunutils", "shotgun_model")
+
+
+def get_item_data(item):
```

Comment:
If you haven't already, want to create a ticket to look into adding additional methods on the sg model and hierarchy model item classes to help out here? maybe make a reference to that future work here? Sounds like we're all in agreement about how to make this better, just unsure of the implementation at this point?